### PR TITLE
Moving monitor outputs used by the server to central storage

### PR DESCRIPTION
**Pull request by york-stsci**
This is the code change part of issue 1395. Before it is merged, all in-use config.json files should be updated to contain a "working" key, whose value should initially be set to the same value as the "outputs" key.

### CODE REVIEW

File: jwql/instrument_monitors/common_monitors/readnoise_monitor.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ def image_to_png(self, image, outname):
             The full path to the output png file.
         """
 
-        output_filename = os.path.join(self.data_dir, '{}.png'.format(outname))
+        output_filename = os.path.join(self.output_data_dir, '{}.png'.format(outname))
```
Comment (BradleySappington) [inline jwql/instrument_monitors/common_monitors/readnoise_monitor.py:256]: change docstring of "data_dir"

File: jwql/instrument_monitors/common_monitors/dark_monitor.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ class Dark():
 
     Attributes
     ----------
-    output_dir : str
+    working_dir : str
```
Comment (BradleySappington) [inline jwql/instrument_monitors/common_monitors/dark_monitor.py]: this doesn't line up with other uses of working dir.  
outputs where outputs are placed (now on central store), working where data is kept and gathered to do calculations.  Is that understanding correct?

File: jwql/instrument_monitors/common_monitors/dark_monitor.py [hunk 3]
```diff
@@ -132,7 +132,7 @@ class Dark():
...
         Path into which outputs will be placed
 
     data_dir : str
```
Comment (BradleySappington) [inline jwql/instrument_monitors/common_monitors/dark_monitor.py]: see above comment, but also, data_dir is renamed to `output_data_dir` in other modules. We should maintain naming conventions.

File: jwql/instrument_monitors/common_monitors/bad_pixel_monitor.py [hunk 4]
```diff
@@ -1110,6 +1110,7 @@ def run(self):
         logging.info('Begin logging for bad_pixel_monitor')
 
         # Get the output directory
+        self.working_dir = os.path.join(get_config()['working'], 'bad_pixel_monitor')
```
Comment (BradleySappington) [inline jwql/instrument_monitors/common_monitors/bad_pixel_monitor.py:1113]: typo bug - should be `working_data_dir`

**Review by BradleySappington — CHANGES_REQUESTED** [review]
A few small changes.  Also please keep naming conventions common through all monitors (when appropriate).  It looks like `working_data_dir` and `output_data_dir` are the most used so would be easiest to stick with that.  I did not mark every instance where naming standard wasn't used.

File: jwql/instrument_monitors/common_monitors/readnoise_monitor.py (hunk 1 above)
Comment (york-stsci) [reply jwql/instrument_monitors/common_monitors/readnoise_monitor.py:256]: Change made.

File: jwql/instrument_monitors/common_monitors/dark_monitor.py (hunk 2 above)
Comment (york-stsci) [reply jwql/instrument_monitors/common_monitors/dark_monitor.py]: That understanding is correct, with one nuance (on which more below).

File: jwql/instrument_monitors/common_monitors/dark_monitor.py (hunk 3 above)
Comment (york-stsci) [reply jwql/instrument_monitors/common_monitors/dark_monitor.py]: Change made.

File: jwql/instrument_monitors/common_monitors/bad_pixel_monitor.py (hunk 4 above)
Comment (york-stsci) [reply jwql/instrument_monitors/common_monitors/bad_pixel_monitor.py:1113]: This is the nuance – some of the common monitors create a "data" directory underneath the working/output directory, usually to separate files by amplifier/detector/whatever, and it's *that* directory that's the working_data_dir. The directory specific to the monitor, by itself, is the working_dir. So essentially, working_data_dir is built from working_dir by appending on /data/ and potentially a further specifier.

Comment (BradleySappington) [reply jwql/instrument_monitors/common_monitors/bad_pixel_monitor.py:1113]: My bad! I just opened up the whole file in a codespace, I see what you're doing now. Apologies for the false alarm!

**Comment by york-stsci** [comment]
@BradleySappington I think I've made all the changes you requested, but I can't figure out how to tell the change request that.

**Review by BradleySappington — APPROVED** [review]

**Comment by BradleySappington** [comment]
I approved, I think that should fix it aside from resolving conflicts and making sure the tests pass

**Pull request merged**